cobbler/items/item.py

```diff
@@ -467,39 +458,91 @@ def _resolve(self, property_name: str) -> Any:
                 f'{type(self)} "{self.name}" does not have property "{property_name}"'
             )
 
-        attribute_value = getattr(self, attribute)
+        return getattr(self, attribute), settings_name
+
+    def __resolve_get_parent_or_settings(self, property_name: str, settings_name: str):
         settings = self.api.settings()
+        if self.parent is not None and hasattr(self.parent, property_name):
+            return getattr(self.parent, property_name)
+        if (
+            hasattr(self, "distro")
+            and getattr(self, "distro") is not None
+            and hasattr(getattr(self, "distro"), property_name)
+        ):
+            # This is only valid for profiles and is a workaround.
+            return getattr(getattr(self, "distro"), property_name)
+        if (
+            hasattr(self, "profile")
+            and getattr(self, "profile") != ""
+            and hasattr(
+                self.api.find_profile(name=getattr(self, "profile")), property_name
+            )
+        ):
+            # This is only valid for systems and is a workaround.
+            return getattr(
+                self.api.find_profile(name=getattr(self, "profile")), property_name
+            )
+        if (
+            hasattr(self, "image")
+            and getattr(self, "image") != ""
+            and hasattr(self.api.find_image(name=getattr(self, "image")), property_name)
+        ):
+            # This is only valid for systems and is a workaround.
+            return getattr(
+                self.api.find_image(name=getattr(self, "image")), property_name
+            )
```

Comment:
Yes, these workarounds are ugly but atm needed. I hope that #3440 can fix this problem. Since without a bigger refactoring I don't see this getting better I think this is the best thing for the interim. It also will be the proper thing for the backport of this PR that you will most likely want to do.

Reply:
Thanks for the PR. It seems doing this while being sick had an effect on me after all... :)

Reply:
It's still not perfect but at least there is only one place to maintain to traverse the hierarchy.